pkg/trustedresources/verify.go

```diff
@@ -45,25 +44,22 @@ const (
 	keyReference = "k8s://"
 )
 
-// VerifyInterface get the checksum of json marshalled object and verify it.
-func VerifyInterface(obj interface{}, verifier signature.Verifier, signature []byte) error {
-	ts, err := json.Marshal(obj)
-	if err != nil {
-		return err
-	}
+var (
+	// DefaultSignatureAlgorithm is the default algorithm for signature
+	DefaultSignatureAlgorithm = "sha256"
 
-	h := sha256.New()
-	h.Write(ts)
-
-	if err := verifier.VerifySignature(bytes.NewReader(signature), bytes.NewReader(h.Sum(nil))); err != nil {
-		return err
+	// supportedSignatureAlgorithms sets a list of support signature algorithms that is similar to the list supported by cosign
+	supportedSignatureAlgorithms = map[string]crypto.Hash{
+		"sha224": crypto.SHA224,
+		"sha256": crypto.SHA256,
+		"sha384": crypto.SHA384,
+		"sha512": crypto.SHA512,
 	}
+)
 
-	return nil
-}
-
-// VerifyTask verifies the signature and public key against task
-func VerifyTask(ctx context.Context, taskObj v1beta1.TaskObject, k8s kubernetes.Interface) error {
+// VerifyTask verifies the signature and public key against task, source is from ConfigSource.URI, which will be used to match policy patterns
+// TODO(#5527): refactor to avoid passing verificationPolicies down the stack.
```

Comment:
I appreciate that the TODO is referring to an issue, however the issue #5527 seems to be encapsulating an awful lot and I think there's a good chance the TODOs could get lost in the mix (and the details will be lost with time). I suggest that either you create separate issues to track each TODO or make the change in this PR.

can you explain a bit more about the change you'd make and why it doesn't make sense to make in this PR? (is this related to the convo with @wlynch  at https://github.com/tektoncd/pipeline/pull/5714#discussion_r1023287564 ?)

Reply:
Oh thanks! This is actually done and I should remove the todo.